--- a/drivers/gpu/drm/tiny/panel-mipi-dbi.c
+++ b/drivers/gpu/drm/tiny/panel-mipi-dbi.c
@@ -248,5 +248,10 @@
 	ret = mipi_dbi_poweron_conditional_reset(dbidev);
+	dev_info(dbidev->drm.dev, "mipi_dbi_poweron_conditional_reset ret=%d", ret);
 	if (ret < 0)
 		goto out_exit;
+	if (!ret)
+		dev_info(dbidev->drm.dev, "ID read failed, will run firmware init");
+	else
+		dev_info(dbidev->drm.dev, "Success read panel ID, skip firmware init");
 	if (!ret)
 		panel_mipi_dbi_commands_execute(dbi, dbidev->driver_private);
