--- a/package/boot/uboot-rockchip/Makefile
+++ b/package/boot/uboot-rockchip/Makefile
@@ -126,6 +126,11 @@ endef
 define U-Boot/hinlink-h28k-rk3528
   $(U-Boot/rk3528/Default)
   NAME:=HINLINK H28K
   BUILD_DEVICES:=hinlink_h28k
 endef

+define U-Boot/hinlink-h29k-rk3528
+  $(U-Boot/rk3528/Default)
+  NAME:=HINLINK H29K
+  BUILD_DEVICES:=hinlink_h29k
+endef
+
@@ -380,6 +385,7 @@ UBOOT_TARGETS := \
        rock-pi-e-v3-rk3328 \
        hinlink-h28k-rk3528 \
+       hinlink-h29k-rk3528 \
        radxa-e20c-rk3528 \
        rock-2-rk3528 \
        nanopi-r3s-rk3566