--- a/Makefile
+++ b/Makefile
@@ -1620,16 +1620,6 @@
-		-a atf-bl1-path=${BL1} \
 		-a atf-bl31-path=${BL31} \
-		-a tee-os-path=${TEE} \
-		-a ti-dm-path=${TI_DM} \
-		-a opensbi-path=${OPENSBI} \
 		-a default-dt=$(default_dt) \
-		-a scp-path=$(SCP) \
 		-a rockchip-tpl-path=$(ROCKCHIP_TPL) \
-		-a spl-bss-pad=$(if $(CONFIG_SPL_SEPARATE_BSS),,1) \
 		-a tpl-bss-pad=$(if $(CONFIG_TPL_SEPARATE_BSS),,1) \
 		-a vpl-bss-pad=$(if $(CONFIG_VPL_SEPARATE_BSS),,1) \
-		-a spl-dtb=$(CONFIG_SPL_OF_REAL) \
-		-a tpl-dtb=$(CONFIG_TPL_OF_REAL) \
-		-a vpl-dtb=$(CONFIG_VPL_OF_REAL) \
-		-a pre-load-key-path=${PRE_LOAD_KEY_PATH} \
 		-a of-spl-remove-props=$(CONFIG_OF_SPL_REMOVE_PROPS) \
