--- a/tools/binman/etype/rockchip_tpl.py
+++ b/tools/binman/etype/rockchip_tpl.py
@@ -18,3 +18,9 @@
     def __init__(self, section, etype, node):
+        fname_arg = "rockchip-tpl-path"
+        filename, = self.GetEntryArgsOrProps([EntryArg(fname_arg, str)])
+        self._hardcoded_tpl_path = filename
         super().__init__(section, etype, node, 'rockchip-tpl')
         self.external = True
+
+    def GetDefaultFilename(self):
+        return self._hardcoded_tpl_path
