--- a/tools/binman/etype/rockchip_tpl.py
+++ b/tools/binman/etype/rockchip_tpl.py
@@ -6,3 +6,4 @@
 from binman.etype.blob_named_by_arg import Entry_blob_named_by_arg
+from u_boot_pylib import tools
 
 class Entry_rockchip_tpl(Entry_blob_named_by_arg):
@@ -18,3 +19,16 @@
     def __init__(self, section, etype, node):
         super().__init__(section, etype, node, 'rockchip-tpl')
         self.external = True
+        self._fixed_abs = "/workdir/openwrt/staging_dir/target-aarch64_generic_musl/image/rk3528_ddr_1056MHz_v1.11.bin"
+
+    def GetDefaultFilename(self):
+        return self._fixed_abs
+
+    @property
+    def _pathname(self):
+        val = self.__dict__.get('_pathname')
+        return val if val is not None else ""
+
+    @_pathname.setter
+    def _pathname(self, value):
+        self.__dict__['_pathname'] = value
