--- a/include/configs/rk3528_common.h
+++ b/include/configs/rk3528_common.h
@@ -20,9 +20,9 @@
 #define ENV_MEM_LAYOUT_SETTINGS			\
 	"scriptaddr=0x00c00000\0"		\
 	"script_offset_f=0xffe000\0"		\
 	"script_size_f=0x2000\0"		\
 	"pxefile_addr_r=0x00e00000\0"		\
-	"kernel_addr_r=0x02000000\0"		\
+	"kernel_addr_r=0x20000000\0"		\
 	"kernel_comp_addr_r=0x0a000000\0"	\
 	"fdt_addr_r=0x12000000\0"		\
 	"fdtoverlay_addr_r=0x12100000\0"	\
